dfm/description/gen_description.py

```diff
@@ -0,0 +1,220 @@
+            cur_label = label_prefix + term
+            out_list.append({cur_label: [{"LOWER": term}]})
+        else:
+            cur_label = label_prefix + label
+            out_list.append({cur_label: [{"LOWER": term}]})
+
+    return out_list
+
+
+def gen_muslim_name_patterns() -> list:
+    from dacy.datasets import muslim_names
+
+    muslim_names_list = [name.lower() for name in muslim_names()["first_name"]]
+
+    return terms_to_lowercase_match_patterns(
+        term_list=muslim_names_list, label="muslim_names"
+    )
+
+
+def gen_gender_name_patterns() -> list:
+    """
+    Creates a list of SpaCy patterns in the shape {"[col_name]": [{"LOWER": "[pattern]"}]}
+    """
+    from dacy.datasets import female_names, male_names
+
+    female_names_list = [name.lower() for name in female_names()["first_name"]]
+    female_names_patterns = terms_to_lowercase_match_patterns(
+        female_names_list, label="female_names"
+    )
+
+    male_names_list = [name.lower() for name in male_names()["first_name"]]
+    male_name_patterns = terms_to_lowercase_match_patterns(
+        male_names_list, label="male_names"
+    )
+
+    return female_names_patterns + male_name_patterns
+
+
+def remove_irrelevant_columns(ds):
+    return ds.remove_columns(["text", "doc_id", "LICENSE", "uri", "date_built"])
+
+
+def gen_matcher_object_from_pattern_list(
+    pattern_container_list: list, nlp: Language
+) -> Matcher:
+    """
+    Generates matcher objects from a list of dictionaries with {matcher_label (str): pattern (list)}
+    Pattern must conform to SpaCy pattern standards:
+
+    Example:
+        >>> pattern_container_list = [
+        >>>    {"atheism": [{"LOWER": {"REGEX": "athei.+"}}]},
+        >>>    {"atheism": [{"LOWER": {"REGEX": "atei.+"}}]},
+        >>>    {"skøde": [{"LOWER": "skøde"}]},
+        >>> ]
+    """
+    matcher_object = Matcher(nlp.vocab)
+
+    for pattern_container in pattern_container_list:
+        pattern_label, subpattern_list = list(*pattern_container.items())
+
+        matcher_object.add(pattern_label, [subpattern_list])
+
+    return matcher_object
+
+
+def get_match_counts_from_doc(doc: Doc, matcher_object: Matcher, nlp: Language) -> dict:
+    """
+    Get match counts for a list of SpaCy matcher-objects
+
+    args:
+        doc (Doc)
+        pattern_container_list (list): A list of dictionaries fitting SpaCy patterns
+        nlp: Language
+
+    returns:
+        A dictionary of the format {pattern_label (str): count (int)}.
+    """
+
+    counts = defaultdict(int)
+
+    # Make sure that all elements are represented in the dict
+    for pattern in matcher_object._patterns:
+        pattern_label = nlp.vocab.strings[pattern]
+
+        counts[pattern_label] = 0
+
+    for match_id, start, end in matcher_object(doc):
+        counts[nlp.vocab.strings[match_id]] += 1
+
+    return dict(counts)
+
+
+def get_match_counts_from_batch(batch, matcher_object: Matcher, nlp: Language) -> dict:
```

Comment:
I would refactor so that this takes texts and create one specifically for HF batches.